src/us/kbase/typedobj/core/TypeProvider.java

```diff
@@ -9,6 +9,26 @@
 /** Provides type definitions to the TypedObjectValidator. */
 public interface TypeProvider {
 
+	/** An error thrown when type information could not be fetched from a remote resource. */ 
+	public static class TypeFetchException extends Exception {
+		private static final long serialVersionUID = 1L;
+
+		/** Create the exception.
+		 * @param message the exception message.
+		 */
+		public TypeFetchException(final String message) {
+			super(message);
+		}
+
+		/** Create the exception.
+		 * @param message the exception message.
+		 * @param cause the cause of the exception.
+		 */
+		public TypeFetchException(final String message, final Throwable cause) {
+			super(message, cause);
+		}
```

Comment:
is this particularly tricky to write a test for? Or are tests coming in a future PR?

Reply:
Future PR. I used to explicitly test exceptions but the code is so trivial it seems pointless, and it's better to test it by using it in a test for a larger piece of code.